File: src/types/service.ts
import { RoomServiceOrder } from './roomServiceOrder';

export interface Service {
    id: number;
    name: string;
    description: string;
    price: number;
    category: string;
    isAvailable: boolean;
    isActive: boolean;
    createdAt: string;
    updatedAt?: string;
    orders?: RoomServiceOrder[];
}

export interface ServiceCreate {
    name: string;
    description: string;
    price: number;
    category: string;
    isAvailable: boolean;
}

export interface ServiceUpdate extends Partial<ServiceCreate> {
    isActive?: boolean;
} 

File: SmartStayHaven.API/Models/Service.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartStayHaven.API.Models
{
    public class Service
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        [Required]
        public decimal Price { get; set; }

        [StringLength(50)]
        public string Category { get; set; }

        public bool IsAvailable { get; set; } = true;

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual ICollection<RoomServiceOrder> Orders { get; set; }
    }
} 